### PR TITLE
Remove Thrift from history service

### PR DESCRIPTION
**Pull request by alexshtin**
Replace all internal usages of `.gen/go/history` and `.gen/go/shared` with proto analogs.

### CODE REVIEW

**Comment by alexshtin** [comment]
All commits are mechanical replacements besides 0a2b921ab5c1f4441539aea019f46ef63234a0f9. @samarabbas let's review it together.

File: service/history/handler.go [hunk 1]
```diff
@@ -1580,42 +1470,120 @@ func (h *Handler) ReapplyEvents(
...
+	return &historyservice.ReapplyEventsResponse{}, nil
 }
 
-// ReadDLQMessages reads replication DLQ messages
-func (h *Handler) ReadDLQMessages(
-	ctx context.Context,
-	request *r.ReadDLQMessagesRequest,
-) (resp *r.ReadDLQMessagesResponse, retError error) {
-	panic("ReadDLQMessages should not be called for thrift handler")
+func (h *Handler) ReadDLQMessages(ctx context.Context, request *historyservice.ReadDLQMessagesRequest) (_ *historyservice.ReadDLQMessagesResponse, retError error) {
```
Comment (samarabbas) [inline service/history/handler.go]: I already moved this entire API to grpc.  I don't think we need the thrift handler definitions for this.

File: service/history/handler.go [hunk 2]
```diff
@@ -1580,42 +1470,120 @@ func (h *Handler) ReapplyEvents(
...
+	return resp, nil
 }
 
-// PurgeDLQMessages deletes replication DLQ messages
-func (h *Handler) PurgeDLQMessages(
-	ctx context.Context,
-	request *r.PurgeDLQMessagesRequest,
-) (retError error) {
-	panic("PurgeDLQMessages should not be called for thrift handler")
+func (h *Handler) PurgeDLQMessages(ctx context.Context, request *historyservice.PurgeDLQMessagesRequest) (_ *historyservice.PurgeDLQMessagesResponse, retError error) {
```
Comment (samarabbas) [inline service/history/handler.go]: I already moved this entire API to grpc.  I don't think we need the thrift handler definitions for this.

File: service/history/handler.go [hunk 3]
```diff
@@ -1580,42 +1470,120 @@ func (h *Handler) ReapplyEvents(
...
+	return &historyservice.PurgeDLQMessagesResponse{}, nil
 }
 
-// MergeDLQMessages reads and applies replication DLQ messages
-func (h *Handler) MergeDLQMessages(
-	ctx context.Context,
-	request *r.MergeDLQMessagesRequest,
-) (resp *r.MergeDLQMessagesResponse, retError error) {
-	panic("MergeDLQMessages should not be called for thrift handler")
+func (h *Handler) MergeDLQMessages(ctx context.Context, request *historyservice.MergeDLQMessagesRequest) (_ *historyservice.MergeDLQMessagesResponse, retError error) {
```
Comment (samarabbas) [inline service/history/handler.go]: I already moved this entire API to grpc.  I don't think we need the thrift handler definitions for this.

File: service/history/handler.go [hunk 4]
```diff
@@ -1580,42 +1470,120 @@ func (h *Handler) ReapplyEvents(
...
+
+	return resp, nil
 }
 
-// RefreshWorkflowTasks refreshes all the tasks of a workflow
-func (h *Handler) RefreshWorkflowTasks(
-	ctx context.Context,
-	request *hist.RefreshWorkflowTasksRequest) (retError error) {
-	panic("RefreshWorkflowTasks should not be called for thrift handler")
+func (h *Handler) RefreshWorkflowTasks(ctx context.Context, request *historyservice.RefreshWorkflowTasksRequest) (_ *historyservice.RefreshWorkflowTasksResponse, retError error) {
```
Comment (samarabbas) [inline service/history/handler.go]: I already moved this entire API to grpc.  I don't think we need the thrift handler definitions for this.

File: service/history/handler.go (hunk 3 above)
Comment (samarabbas) [reply service/history/handler.go]: Just making sure you completely removed the thrift handlers for history service and now only handler we register is grpc based?

File: service/history/handler.go (hunk 2 above)
Comment (samarabbas) [reply service/history/handler.go]: Just making sure you completely removed the thrift handlers for history service and now only handler we register is grpc based?

File: service/history/handler.go (hunk 1 above)
Comment (samarabbas) [reply service/history/handler.go]: Just making sure you completely removed the thrift handlers for history service and now only handler we register is grpc based?

File: service/history/handler.go (hunk 4 above)
Comment (samarabbas) [reply service/history/handler.go]: Just making sure you completely removed the thrift handlers for history service and now only handler we register is grpc based?

File: service/history/handler.go [hunk 5]
```diff
@@ -154,21 +159,26 @@ func (h *Handler) CreateEngine(
 	)
 }
 
+// Health is for health check
+func (h *Handler) Health(context.Context, *healthservice.HealthRequest) (_ *healthservice.HealthStatus, retError error) {
+	h.startWG.Wait()
+	h.GetLogger().Debug("Matching service health check endpoint (gRPC) reached.")
```
[the remark this answers is not in the document]
Comment (alexshtin) [reply service/history/handler.go]: image

File: service/history/handler.go (hunk 1 above)
Comment (alexshtin) [reply service/history/handler.go]: `handler.go` is gRPC now. I removed `handlerGRPC.go` completely.

**Pull request merged**